shell/app/node_main.cc

```diff
@@ -64,6 +111,9 @@ int NodeMain(int argc, char* argv[]) {
     // Explicitly register electron's builtin modules.
     NodeBindings::RegisterBuiltinModules();
 
+    // Parse and set Node.js cli flags.
+    SetNodeCliFlags();
+
     int exec_argc;
     const char** exec_argv;
     node::Init(&argc, const_cast<const char**>(argv), &exec_argc, &exec_argv);
```

Comment:
Can't we have `node::Init` do the call to `ProcessGlobalArgs` since it has access to commandline and also we are allowing all node cli flags for this mode, it only makes sense for node to handle rather than us.

Only reason we want it here is for the crypto related flags, what happens if we let them through to node ? We don't build with those crypto config support, so ideally node should error in those situations which is acceptable behavior as an embedder.

Reply:
We don’t just want it here for crypto-related flags - without explicitly setting these from our command line they simply don’t get set. `ProcessGlobalArgs` doesn’t have access to the command line - it just parses what we pass to it via `argv`

Reply:
`argv` passed to `node::Init` is the same one `base::CommandLine` uses, check the start of this function, why would they not be available to node to process further ?

Reply:
Right, i see that, but do you feel it's _bad_  to save users from experiencing what will be deterministic and unexpected failures?

Reply:
What does node throw at the moment when we don't handle the parsing ? Just an exit code and no descriptive erro ?

Reply:
no - it'll fail when the user then goes to invoke the crypto-related methods. Since we're bundling BoringSSL, which is not always clear to users, I personally feel that it's fair for us to assume responsibility for helping users not run into failing functionality we're already clearly aware will be failing when they go to invoke it.